smart-contracts/test/Lock/destroyLock.js

```diff
@@ -1,123 +1,177 @@
 const Units = require('ethereumjs-units')
 const BigNumber = require('bignumber.js')
 
+const Web3Utils = require('web3-utils')
 const deployLocks = require('../helpers/deployLocks')
 const shouldFail = require('../helpers/shouldFail')
+const LockApi = require('../helpers/lockApi')
+const getTokenBalance = require('../helpers/getTokenBalance')
 
 const unlockContract = artifacts.require('../Unlock.sol')
 const getUnlockProxy = require('../helpers/proxy')
 
+const TestErc20Token = artifacts.require('TestErc20Token.sol')
+
 let unlock, locks
 
 contract('Lock / destroyLock', accounts => {
   let lock
+  let lockApi
+  let testToken
+  const scenarios = [false, true]
 
   before(async () => {
-    unlock = await getUnlockProxy(unlockContract)
-    locks = await deployLocks(unlock, accounts[0])
-    lock = locks['FIRST']
+    testToken = await TestErc20Token.new()
+    // Mint some tokens for testing
+    for (let i = 0; i < accounts.length; i++) {
+      await testToken.mint(accounts[i], '1000000000000000000', {
+        from: accounts[0],
+      })
+    }
   })
 
-  it('should fail if called by the wrong account', async () => {
-    await shouldFail(lock.destroyLock({ from: accounts[1] }), '')
-  })
+  scenarios.forEach(isErc20 => {
+    describe(`Test ${isErc20 ? 'ERC20' : 'ETH'}`, () => {
+      let tokenAddress
+      before(async () => {
+        tokenAddress = isErc20 ? testToken.address : Web3Utils.padLeft(0, 40)
+        unlock = await getUnlockProxy(unlockContract)
+        locks = await deployLocks(unlock, accounts[0], tokenAddress)
+        lock = locks['FIRST']
+        lockApi = new LockApi(lock)
+
+        for (let i = 0; i < accounts.length; i++) {
+          await testToken.approve(lock.address, -1, { from: accounts[i] })
+        }
 
-  describe('when called by the owner', () => {
-    let initialLockBalance, initialOwnerBalance, txObj, event
+        // Add ETH to the lock, even if it's priced in ERC20
+        // TODO: should we block this from happening instead?
```

Comment:
As long as `destroyLock` sends ether and/or erc20 tokens to owner it's probably fine.